app/routes/privateRoutes/restClientPage/serverFetch.ts

```diff
@@ -0,0 +1,54 @@
+import { type ActionFunctionArgs } from 'react-router';
+
+import { HttpMethods, Protocols } from '@/sources/enums';
+
+import { getFinalUrlParams } from '@/utils/fetch/getFinalUrlParams';
+import { mergedDataResponse } from '@/utils/fetch/mergeDataResponse';
+
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const formData = await request.formData();
+  const data = Object.fromEntries(formData);
+
+  const protocol = data.protocol || Protocols.HTTPS;
+  const url = protocol + (data.url as string);
```

Comment:
```suggestion
  const url = `${protocol} ${data.url}`;
```
лучше строки не конкатенировать